charts/rollups-validator-node/templates/deployment.yaml

```
@@ -3,6 +3,101 @@
 SPDX-License-Identifier: Apache-2.0 (see LICENSE)
 */}}
 
+{{- if .Values.validator.localnode.enabled }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "validator.fullname" . }}-localnode
+  labels: {{- include "validator.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "validator.selectorLabels" . | nindent 6 }}
+      rollups.cartesi.io/component: localnode
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      {{- with .Values.validator.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "validator.selectorLabels" . | nindent 8 }}
+        rollups.cartesi.io/component: localnode
+    spec:
+      {{- with .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "validator.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.validator.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: anvilstate
+          image: sunodo/rollups-node:0.2.0
+          command:
+            - cp
+            - /opt/cartesi/share/deployments/anvil_state.json
+            - /opt/cartesi/share/deployments/localhost.json
+            - /opt/anvil/
+          volumeMounts:
+            - name: blockchain-data
+              mountPath: /opt/anvil
+              readOnly: false
+      containers:
+        - name: anvil
+          image: {{ include "anvil.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          resources:
+            {{- toYaml .Values.validator.localnode.anvil.resources | nindent 12 }}
+          {{- if .Values.validator.localnode.anvil.command }}
+          command:
+          {{- include "tplvalues.render" (dict "value" .Values.validator.localnode.anvil.command "context" $) | nindent 12 }}
+          {{- else }}
+          args:
+            - anvil
+            - --block-time
+            - "5"
+            - --load-state
+            - /opt/cartesi/share/deployments/anvil_state.json
+            {{- end }}
```

```suggestion
          {{- end }}
```

Is this aligned/indented?

---

My bad, Fixed !

---

not yet, I think

https://github.com/cartesi/helm-charts/pull/9/files#diff-32350292ed0339c09e8a4fb34d642f6c085ba42a64f1548e2a0d4e87249530b0R43

---

https://github.com/cartesi/helm-charts/pull/9/files#diff-3bf9ac3c0269367d8bfddf5e388ea7039af8025334f92d84d9a9655366a6fdbfR66

---

if the user defines the `validator.localnode.anvil.command` they won't be able to also define `validator.localnode.anvil.args` not `validator.localnode.anvil.extraArgs`

we need a better logic here

---

moved the args to values.yaml like `serverManager`

---

So the caveat here is, if the user wants a different command, they will have to inform some `args:`  or make it empty.

We'd have to check if it's consistent with all other services.

I'll approve it as is so we can move forward.